lib/hydra/works/services/add_file_to_file_set.rb

```diff
@@ -76,6 +76,7 @@ def find_or_create_file_for_rdf_uri(type, update_existing)
             current_file = file_set.files.last
             Hydra::PCDM::AddTypeToFile.call(current_file, type_to_uri(type))
           end
+          current_file
```

Comment:
Yes, tl;dr - it ensures the return value of `current_file` (instead of `nil` as we were getting at first).

Consider this simplified logic:
```ruby
def foo(x)
  y = "x is true" if x
  unless x
    "x is not true"
  end
end

foo(false)
# => "x is false"
foo(true)
# => nil
```

So... we started by copying `#find_or_create_file_for_rdf_uri` from `AddFileToFileSet::Updater` to our new `AddExternalFileToFileSet::Updater`, without the explicit return value.

Then, in our testing context, we were adding new external file URLs to the same `FileSet` objects with the same RDF type.

The result, is that the first assignment worked, but the method was returning the results of the `unless` block, which is `nil` in that case. That value then get's assigned to the `@current_file` attribute of the `Updater` class, which broke stuff.

The takeaway is that `#find_or_create_file_for_rdf_uri` needs to either return a `File` object, or raise an error, but we found a case (in our tests) where it was returning `nil`.

Explicitly specifying the return value helps prevent bugs that we haven't discovered yet, because there are currently no unit tests around `AddFileToFileSet::Updater`. I would be in favor of adding them, but I was hoping that could be a separate ticket, since it's only tangentially related to what we're actually trying to add here.